controllers/configurationpolicy_controller.go

```diff
@@ -2711,10 +2741,58 @@ func (r *ConfigurationPolicyReconciler) checkAndUpdateResource(
 
 		updateSucceeded = true
 	} else {
+		if throwSpecViolation && recordDiff != policyv1.RecordDiffNone {
+			// The spec didn't require a change but throwSpecViolation indicates the status didn't match. Handle
+			// this diff for this case.
+			mergedObjCopy := obj.existingObj.DeepCopy()
+			removeFieldsForComparison(mergedObjCopy)
+
+			// The provided isInform value is always true because the status checking can only be inform.
+			diff = handleDiff(log, recordDiff, true, existingObjectCopy, mergedObjCopy)
+		}
+
 		r.setEvaluatedObject(obj.policy, obj.existingObj, !throwSpecViolation)
 	}
 
-	return throwSpecViolation, "", updateNeeded, updateSucceeded
+	return throwSpecViolation, "", diff, updateNeeded, updateSucceeded
+}
+
+// handleDiff will generate the diff and then log it or return it based on the input recordDiff value. If recordDiff
+// is set to None or is set to InStatus with enforce, no diff is generated. When recordDiff is set to Sensored, a
```

Comment:
inform + inStatus creates diff?  why?

Reply:
If the policy is enforced, then the diff is no longer relevant after the update so it shouldn't get stored in the status.

Reply:
I was confused about this as well, can the comment explain that detail?

Reply:
I was confused about this as well, can the comment explain that detail?

Reply:
I added a comment that should help clarify this.